src/Accessor/DefaultAccessorStrategy.php

```diff
@@ -20,21 +20,47 @@
 
 namespace JMS\Serializer\Accessor;
 
+use GeneratedHydrator\Configuration;
+use JMS\Serializer\Metadata\ClassMetadata;
 use JMS\Serializer\Metadata\PropertyMetadata;
 
 /**
  * @author Asmir Mustafic <goetas@gmail.com>
  */
 final class DefaultAccessorStrategy implements AccessorStrategyInterface
 {
+    private $gen = array();
 
-    public function getValue(object $object, PropertyMetadata $metadata)
+    public function getValue(object $object, PropertyMetadata $metadata, $context)
     {
+
+        if (!$metadata->getter && array_key_exists($metadata->name, $context)) {
+            return $context[$metadata->name];
+        }
+
         return $metadata->getValue($object);
     }
 
     public function setValue(object $object, $value, PropertyMetadata $metadata): void
     {
         $metadata->setValue($object, $value);
     }
+
+    public function startAccessing(object $object, ClassMetadata $metadata)
+    {
+        $class = get_class($object);
+
+        if (!isset($this->gen[$class])) {
+            $config = new Configuration($class);
+            $hydratorClass = $config->createFactory()->getHydratorClass();
+            $this->gen[$class] = new $hydratorClass();
+        }
+
+        return $this->gen[$class]->extract($object);
```

Comment:
@Majkl578 generators are generated and cached, so is "production" ready after the warmup.

Reply:
The code inside the "if" is never executed during the benchmark, so should not make it slower